test/object_test.go

```diff
@@ -486,27 +520,86 @@ func TestObjectLinks(t *testing.T) {
 	_, err = dir.PutString("DIR/B", "DIR-BBB")
 	expectOk(t, err)
 
-	info, err = dir.GetInfo("DIR/B")
+	infoB, err := dir.GetInfo("DIR/B")
 	expectOk(t, err)
 
-	binfo, err := root.AddLink("DBL", info)
+	infoLB, err := root.AddLink("DBL", infoB)
 	expectOk(t, err)
-
-	if binfo.Name != "DBL" || binfo.NUID == "" || binfo.ModTime.IsZero() {
-		t.Fatalf("Link info not what was expected: %+v", binfo)
-	}
+	expectLinkIsCorrect(t, infoB, infoLB)
 
 	// Now add whole other store as a link, like a directory.
-	_, err = root.AddBucketLink("dir", dir)
+	infoBucketLink, err := root.AddBucketLink("dir", dir)
 	expectOk(t, err)
 
-	// Now try to get a linked object.
-	dbl, err := root.GetString("DBL")
+	expectLinkPartsAreCorrect(t, infoBucketLink, "DIR", "")
+
+	// Try to get a linked object, same bucket
+	getLA, err := root.GetString("LA")
 	expectOk(t, err)
 
-	if dbl != "DIR-BBB" {
-		t.Fatalf("Expected %q but got %q", "DIR-BBB", dbl)
+	if getLA != "AAA" {
+		t.Fatalf("Expected %q but got %q", "AAA", getLA)
+	}
+
+	// Test the link to a link get
+	//getLALA, err := root.GetString("LALA")
+	//expectOk(t, err)
+	//
+	//if getLALA != "AAA" {
+	//	t.Fatalf("Expected %q but got %q", "AAA", getLALA)
+	//}
+
+	// Try to get a linked object, cross bucket
+	getDbl, err := root.GetString("DBL")
+	expectOk(t, err)
+
+	if getDbl != "DIR-BBB" {
+		t.Fatalf("Expected %q but got %q", "DIR-BBB", getDbl)
+	}
+
+	// Check simple errors.
+	_, err = root.AddLink("", infoB)
+	expectErr(t, err)
+
+	_, err = root.AddLink("Nil Object", nil)
+	expectErr(t, err)
+
+	infoB.Name = ""
+	_, err = root.AddLink("Empty Info Name", infoB)
+	expectErr(t, err)
+
+	// Check Error Link to a Link
+	_, err = root.AddLink("Link To Link", infoLB)
+	expectErr(t, err)
+
+	// Check Error Link to a Link
+	_, err = root.AddBucketLink("", root)
+	expectErr(t, err)
+
+	_, err = root.AddBucketLink("Nil Bucket", nil)
+	expectErr(t, err)
+}
+
+func expectLinkIsCorrect(t *testing.T, originalObject *nats.ObjectInfo, linkObject *nats.ObjectInfo) {
```

Comment:
Usually in go you check for error conditions first and if no error, continue execution. Also, we can easily avoid nesting if blocks here. So how about:
```
func expectLinkIsCorrect(t *testing.T, originalObject *nats.ObjectInfo, linkObject *nats.ObjectInfo) {
	if linkObject.Opts.Link == nil || !expectLinkPartsAreCorrect(t, linkObject, originalObject.Bucket, originalObject.Name) {
		t.Fatalf("Link info not what was expected:\nActual: %+v\nTarget: %+v", linkObject, originalObject)
	}
}
```